File: Proyecto/02. Software/02. Back-End Java/workspace/sell.it.easy/src/main/java/com/sell/it/easy/entity/Usuario.java
package com.sell.it.easy.entity;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Lob;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;

import com.fasterxml.jackson.annotation.JsonFormat;

@Entity
public class Usuario {
	
	@Id
	@GeneratedValue(strategy=GenerationType.AUTO)
	private Integer id;
	
	@Column(nullable=false)
	private String nombre;
	
	private String apellidos;
	
	@Column(nullable=false)
	private String correo;
	
	@Column(nullable=false)
	private String password;
	
	@ManyToMany(cascade=CascadeType.ALL, fetch=FetchType.EAGER)
	private List<Rol> roles = new ArrayList<Rol>();
	
	@Lob
	private String imagenString;
	
	@JsonFormat(pattern="dd/MM/yyyy HH:mm")
	private Date fechaNacimiento;
	
	public Usuario(){}
	
	
	public Integer getId() {
		return id;
	}
	public void setId(Integer id) {
		this.id = id;
	}
	public String getNombre() {
		return nombre;
	}
	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
	public String getApellidos() {
		return apellidos;
	}
	public void setApellidos(String apellidos) {
		this.apellidos = apellidos;
	}
	public String getCorreo() {
		return correo;
	}
	public void setCorreo(String correo) {
		this.correo = correo;
	}
	public String getPassword() {
		return password;
	}
	public void setPassword(String password) {
		this.password = password;
	}


	public List<Rol> getRoles() {
		return roles;
	}


	public void setRoles(List<Rol> roles) {
		this.roles = roles;
	}
	
	public void addRoles(Rol rol) {
		this.roles.add(rol);
	}

	public String getImagenString() {
		return imagenString;
	}


	public void setImagenString(String imagenString) {
		this.imagenString = imagenString;
	}


	public Date getFechaNacimiento() {
		return fechaNacimiento;
	}


	public void setFechaNacimiento(Date fechaNacimiento) {
		this.fechaNacimiento = fechaNacimiento;
	}
	
	
}

File: Proyecto/02. Software/01. Front-End iOS/sell.it.easy_0.0.2/sell.it.easy/sell.it.easy/Usuario.swift
//
//  Usuario.swift
//  sell.it.easy
//
//  Created by David Márquez Delgado on 13/8/17.
//  Copyright © 2017 David Márquez Delgado. All rights reserved.
//

import Foundation
import ObjectMapper

class Usuario: Mappable {
    var id : Int?
    var correo : String?
    var nombre : String?
    var apellidos : String?
    var password : String?
    var imagenString : String?
    var roles = [Rol]()
    
   init(id: Int?, correo: String?, nombre: String?, apellidos: String?, password: String?, imagenString: String?) {
        self.id = id
        self.correo = correo
        self.nombre = nombre
        self.apellidos = apellidos
        self.password = password
        self.imagenString = imagenString
    }
    
    required init?(map: Map) {}
    
    func mapping(map: Map) {
        id <- map["id"]
        correo <- map["correo"]
        nombre <- map["nombre"]
        apellidos <- map["apellidos"]
        password <- map["password"]
        imagenString <- map["imagenString"]
        roles <- map["roles"]
    }
}
